src/mpid/ch3/channels/nemesis/netmod/portals4/ptl_send.c

```diff
@@ -161,15 +161,15 @@ static int send_msg(ptl_hdr_data_t ssend_flag, struct MPIDI_VC *vc, const void *
         
         /* noncontig data */
         MPL_DBG_MSG(MPIDI_CH3_DBG_CHANNEL, VERBOSE, "Small noncontig message");
-        sreq->dev.segment_ptr = MPIR_Segment_alloc();
+        sreq->dev.segment_ptr = MPIR_Segment_alloc(buf, count, datatype);
         MPIR_ERR_CHKANDJUMP1(sreq->dev.segment_ptr == NULL, mpi_errno, MPI_ERR_OTHER, "**nomem", "**nomem %s", "MPIR_Segment_alloc");
-        MPIR_Segment_init(buf, count, datatype, sreq->dev.segment_ptr);
         sreq->dev.segment_first = 0;
         sreq->dev.segment_size = data_sz;
 
         last = sreq->dev.segment_size;
         sreq->dev.iov_count = MPL_IOV_LIMIT;
         MPIR_Segment_to_iov(sreq->dev.segment_ptr, sreq->dev.segment_first, &last, sreq->dev.iov, &sreq->dev.iov_count);
+        MPIR_Segment_free(sreq->dev.segment_ptr);
 
```

Comment:
Looks like you fixed the unreleased segment issue in portals. Could you please also fix the same issue in mxm ? Besides, this fix should be put into a separate commit.

Reply:
Done.